### PR TITLE
UGENE-7572 QT error in log after cancelling PHY tree creation.

### PR DESCRIPTION
**Pull request by rasputinkirill**

### CODE REVIEW

**Review by yalgaer — COMMENTED** [review]
Looks good please fix all builds and run GUI tests.

File: src/corelibs/U2Core/src/tasks/ExternalToolRunTask.cpp [hunk 1]
```diff
@@ -120,7 +120,12 @@ void ExternalToolRunTask::run() {
     while (!externalToolProcess->waitForFinished(1000)) {
         if (isCanceled()) {
             killProcess(externalToolProcess);
-            algoLog.details(tr("Tool %1 is cancelled").arg(toolName));
+            externalToolProcess->waitForFinished(10000);
+            if (externalToolProcess->state() == QProcess::Running) {
+                algoLog.info(tr("Unable to cancel tool %1 for 10 seconds. Stop it manually by your OS task manager.").arg(toolName));
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Core/src/tasks/ExternalToolRunTask.cpp:127]: Wait a bit, you can try to kill it by yourself, just by using [QProcess::kill()](https://doc.qt.io/qt-5/qprocess.html#kill). I'm sure you should try to do it and if you failed, then you text this message to user

**Review by DmitriiSukhomlinov — CHANGES_REQUESTED** [review]
See the comment

File: src/corelibs/U2Core/src/tasks/ExternalToolRunTask.cpp [hunk 2]
```diff
@@ -120,7 +120,16 @@ void ExternalToolRunTask::run() {
     while (!externalToolProcess->waitForFinished(1000)) {
         if (isCanceled()) {
             killProcess(externalToolProcess);
-            algoLog.details(tr("Tool %1 is cancelled").arg(toolName));
+            externalToolProcess->waitForFinished(2000); //wait 2 seconds to let OS kill process in previous method
+            if (externalToolProcess->state() == QProcess::Running) {
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Core/src/tasks/ExternalToolRunTask.cpp]: Check here the return value of `waitForFinished()`, not the state

Comment (DmitriiSukhomlinov) [reply src/corelibs/U2Core/src/tasks/ExternalToolRunTask.cpp]: Also, you should check `externalToolProcess->exitStatus()` - it should be `QProcess::NormalExit`

File: src/corelibs/U2Core/src/tasks/ExternalToolRunTask.cpp [hunk 3]
```diff
@@ -120,7 +120,16 @@ void ExternalToolRunTask::run() {
...
+                externalToolProcess->kill();
+            }
+            externalToolProcess->waitForFinished(10000);
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Core/src/tasks/ExternalToolRunTask.cpp]: The same checks

**Review by DmitriiSukhomlinov — CHANGES_REQUESTED** [review]
See comments

**Review by DmitriiSukhomlinov — APPROVED** [review]

**Review by yalgaer — APPROVED** [review]

**Pull request merged**